### PR TITLE
Fixes yiisoft/yii2#10710

### PR DESCRIPTION
**Pull request by wallysalami**
Fixes yiisoft/yii2#10710

### CODE REVIEW

File: framework/helpers/BaseFileHelper.php [hunk 1]
```diff
@@ -253,6 +253,9 @@ protected static function loadMimeTypes($magicFile)
      */
     public static function copyDirectory($src, $dst, $options = [])
     {
+        if ($src === $dst || strpos($dst, $src) !== false){
```
Comment (cebe) [inline framework/helpers/BaseFileHelper.php]: strpos should only match if its result is 0, e.g. copying from `/data` to `/backup/data` should work.
Would be good to also add tests to https://github.com/yiisoft/yii2/blob/master/tests/framework/helpers/FileHelperTest.php

Comment (SilverFire) [inline framework/helpers/BaseFileHelper.php]: `){` -> `) {`

File: framework/helpers/BaseFileHelper.php [hunk 2]
```diff
@@ -253,6 +253,9 @@ protected static function loadMimeTypes($magicFile)
...
+            throw new InvalidParamException( "Trying to copy a directory to itself or a subdirectory." );
```
Comment (SilverFire) [inline framework/helpers/BaseFileHelper.php]: Could you please remove redundant spaces before after `(` and before `)`

**Comment by SilverFire** [comment]
Merged https://github.com/yiisoft/yii2/commit/4f4fa6fdf0a45921ef48ffcd2941352c33a8dd28

**Pull request closed without merging**